### PR TITLE
Update Registry DetailsSubInfo to include Dev Rewards for Agents & Components

### PR DESCRIPTION
**Pull request by truemiller**
## Proposed changes

Update Registry DetailsSubInfo to include Dev Rewards for Agents & Components

| No rewards comp | Rewards comp | Rewards agent |
|--------|--------|--------|
| image| image| image| 

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: apps/autonolas-registry/common-util/Details/DetailsSubInfo.jsx [hunk 1]
```diff
@@ -82,6 +101,30 @@ ViewHashAndCode.defaultProps = {
...
+ * Displays rewards earned and rewards top up
+ */
+const RewardsSection = ({ reward, topUp }) => {
+  return (
+    <Row>
+      <Col span={24} xl={12}>
+        <RewardsStatistic title={'Claimable Reward'} value={reward} suffix="ETH" />
+      </Col>
+      <Col span={24} xl={12}>
+        <RewardsStatistic title="Claimable Top Up" value={topUp} suffix="OLAS" />
```
Comment (oaksprout) [inline apps/autonolas-registry/common-util/Details/DetailsSubInfo.jsx:114]: 4 decimal places works for ETH, but for OLAS I suggest 2 decimal places.

**Review by oaksprout — APPROVED** [review]

**Comment by truemiller** [comment]
tests are failing on registry as there's no textencoder for viem in tests, so there a preferred formatter i should be using here @mohandast52 @Tanya-atatakai ?

**Comment by Tanya-atatakai** [comment]
>  there a preferred formatter i should be using here
@truemiller I don't think there is, I've got this error too. I think the problem is in the messed up here and there jest configs + ts configs, I'm gonna look into it today and hopefully find a solution. I'd tried to specify the text encoder but then new issues appeared, so no luck

**Comment by truemiller** [comment]
Have included TextEncoder in jest setup to get around the errors thrown by viem. A bunch of tests seem to be broken/outdated however.

File: apps/autonolas-registry/common-util/Details/index.tsx [hunk 2]
```diff
@@ -1,34 +1,36 @@
...
+  navigateToDependency?: (id: string, type: NavTypesValues) => void,
   renderServiceState?: (props: {
-    isOwner: boolean;
-    details: GenericObject;
-    updateDetails: (details: GenericObject) => void;
-  }) => JSX.Element | null;
+    isOwner: boolean,
+    details: GenericObject,
+    updateDetails: (details: GenericObject) => void,
+  }) => JSX.Element | null,
```
[the remark this answers is not in the document]
Comment (truemiller) [reply apps/autonolas-registry/common-util/Details/index.tsx:33]: not sure what happened here...

**Comment by truemiller** [comment]
> > A bunch of tests seem to be broken/outdated however.
> 
> @truemiller, Curious to know which tests specifically? They should be up to date and at least cover the details and list page.

quite a few, including ones that seem unfinished.
happy to resolve but would prefer to do this in another ticket as this one's been open for quite a while for a small feature now

**Pull request merged**